samples/java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCartImpl.java

```diff
@@ -16,94 +16,109 @@
 
 package com.example.shoppingcart.domain;
 
+import com.akkaserverless.javasdk.Effect;
 import com.akkaserverless.javasdk.EntityId;
 import com.akkaserverless.javasdk.eventsourcedentity.CommandContext;
 import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntity;
 import com.example.shoppingcart.ShoppingCartApi;
 import com.google.protobuf.Empty;
 
-import java.util.LinkedHashMap;
-import java.util.Map;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 @EventSourcedEntity(entityType = "eventsourced-shopping-cart")
```

Comment:
Where should the entityType be defined? I think it belongs to the registration, similar to how the viewId is defined in the registration of views.

Reply:
In the registration sounds good. There may be an option for that already? (I remember this being discussed before)

Reply:
since we are touching all this, it might be good to rename this to `entityTypeHint`. Maybe on a follow-up PR.